apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -0,0 +1,852 @@
+     */
+    _onDefaultAdapterChanged:
+    function btc__onDefaultAdapterChanged(newAdapter, oldAdapter) {
+      this.debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+      this.debug('_onDefaultAdapterChanged(): oldAdapter = ' + oldAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+
+      if (oldAdapter) {
+        // unwatch event since the old adapter is no longer usefull
+        this._unwatchDefaultAdapterOnattributechanged(oldAdapter);
+        this._unwatchDefaultAdapterOndeviceunpaired(oldAdapter);
+        this._unwatchDefaultAdapterOndevicepaired(oldAdapter);
+      }
+
+      if (newAdapter) {
+        // watch event since the new adapter is ready to access
+        this._initProperties(newAdapter);
+        this._watchDefaultAdapterOnattributechanged(newAdapter);
+        this._watchDefaultAdapterOndevicepaired(newAdapter);
+        this._watchDefaultAdapterOndeviceunpaired(newAdapter);
+      } else {
+        // reset properties only
+        this._resetProperties();
+      }
+    },
+
+    /**
+     * State of init for paired devices info. We only init paired devices info
+     * one time while _init() or 'onattributechanged' event comimg from default
+     * adapter.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @type {Boolean}
+     */
+    _hasInitPairedDevicesInfo: false,
+
+    /**
+     * Init paired devices information.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object} adapter
+     */
+    _initPairedDevicesInfo: function btc__initPairedDevicesInfo(adapter) {
+      if (!this._hasInitPairedDevicesInfo) {
+        this._hasInitPairedDevicesInfo = true;
+        this._refreshPairedDevicesInfo(adapter);
+      }
+    },
+
+    /**
+     * Refresh paired devices information.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object BluetoothAdapter} adapter
+     */
+    _refreshPairedDevicesInfo: function btc__refreshPairedDevicesInfo(adapter) {
+      var pairedDevices = adapter.getPairedDevices();
+      this.debug('pairedDevices.length = ' + pairedDevices.length);
+
+      // reset paired devices list
+      this._pairedDevices.reset([]);
+
+      // refresh properties about paired devices
+      if (pairedDevices.length === 0) {
+        // reset the name
+        this.firstPairedDeviceName = '';
+        // update property 'hasPairedDevice'
+        this.hasPairedDevice = false;
+      } else {
+        // save paired devices in list
+        this._getBluetoothDevice().then((BtDevice) => {
+          for (var i in pairedDevices) {
+            // create observable BtDevice
+            var observableBtDevice = BtDevice(pairedDevices[i]);
+            // push device in devices list with observable object
+            this._pairedDevices.push(observableBtDevice);
+          }
+        });
+
+        // copy for sorting
+        var paired = pairedDevices.slice();
```

Comment:
If we remove `getBluetoothDevice`, we don't have to make a copy of `pairedDevice` as you already push them to `this._pairedDevices`.